packages/blocks/src/api/test/registration.js

```diff
@@ -632,6 +632,31 @@ describe( 'blocks', () => {
 					);
 				} );
 			} );
+
+			it( 'should update block attributes separately for each block when they use a default set', () => {
+				addFilter(
+					'blocks.registerBlockType',
+					'core/blocks/shared-defaults',
+					( settings, name ) => {
+						if ( name === 'my-plugin/test-block-1' ) {
+							settings.attributes.newlyAddedAttribute = {
```

Comment:
What happens if you avoid mutations?

Reply:
Things will work well, as described in the linked issue. I have written all my filters avoiding mutation just from a habit so I have never discovered this situation but I think we should just play it safe and prevent the possibility. 

This would be ok:

```
settings.attributes = {
  ...settings.attributes,
  newlyAddedAttribute: {},
};
```

What do you think? I'm thinking about removing the declaration of `DEFAULT_BLOCK_TYPE_SETTINGS` and moving the default attributes directly inside `registerBlockType` so each block gets their own instance. Or keep the constant and do a deep copy before using it which I probably like better

Reply:
Agreed, we should still recommend immutability but we should clone the defaults

Reply:
I think this is ready now